src/conn/conn_api.c

```diff
@@ -1055,9 +1055,10 @@ __conn_close(WT_CONNECTION *wt_conn, const char *config)
 	if (cval.val != 0)
 		F_SET(conn, WT_CONN_LEAK_MEMORY);
 	WT_TRET(__wt_config_gets(session, cfg, "use_timestamp", &cval));
-	if (cval.val != 0 && conn->txn_global.has_stable_timestamp) {
+	if (cval.val != 0) {
 		ckpt_cfg = "use_timestamp=true";
-		F_SET(conn, WT_CONN_CLOSING_TIMESTAMP);
+		if (conn->txn_global.has_stable_timestamp)
+			F_SET(conn, WT_CONN_CLOSING_TIMESTAMP);
 	}
```

Comment:
I'm slightly confused by this change -- what does it mean to set `use_timestamp=true` if there is no `stable_timestamp`?

Reply:
Okay, I understand the intent -- as we've discussed previously, I feel like we're twisting the checkpoint interface here in a way I find counterintuitive (it would make more sense to me to preserve the last timestamp at which a checkpoint completed and have MongoDB explicitly clear that via some version of `WT_CONNECTION::set_timestamp`).

That's not the subject of this change, though, and something we could revisit in future.